### PR TITLE
Fix Formula for Ohio Citizen Credits

### LINKED ISSUE #3175: Fix formula for Ohio senior citizen credit
In the 0.507.1 version, the specification of the formula for the `oh_senior_citizen_credit` variable causes **the model to crash** with the following error:
```
variables/gov/states/oh/tax/income/credits/oh_senior_citizen_credit.py
line 29, in formula
    any_elderly * has_not_taken_lump_sum_distribution * credit_amount
ValueError: operands could not be broadcast together
            with shapes (100000,) (279905,) 
```
This error occurs because `any_elderly` and `credit_amount` are tax_unit variables, but
`has_not_taken_lump_sum_distribution` is a person variable.

### PR DESCRIPTION
**Pull request by vrathi101**
### Summary
:memo::bug::wastebasket:

This pull request fixes the Ohio senior citizen credit formula in the policy engine by using the person level instead of the tax unit level. It also updates the changelog file with a new entry for the minor version increment.

> _`Ohio` credit_
> _fixed with person level logic_
> _old code commented_

### Walkthrough
*  Update the Ohio senior citizen credit formula to use the person level instead of the tax unit level ([link](https://github.com/PolicyEngine/policyengine-us/pull/3238/files?diff=unified&w=0#diff-00f5c4d4ff65ddd24e737d574db58a163fe8a93d7b7e4567b8fc9b3afb05900cL25-R33))
*  Add a new entry to the changelog file to document the version increment and the bug fix ([link](https://github.com/PolicyEngine/policyengine-us/pull/3238/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8L1-R3))

Fixes #3175

### CODE REVIEW

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need a git pull upstream master

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    fixes:
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_senior_citizen_credit.py [hunk 2]
```diff
@@ -20,11 +20,12 @@ def formula(tax_unit, period, parameters):
         has_not_taken_lump_sum_distribution = person(
             "oh_has_not_taken_oh_lump_sum_credits", period
         )
-        age = person("age", period)
 
-        any_elderly = tax_unit.any(age >= p.age_threshold)
+        age = person("age", period)
+        any_elderly = age >= p.age_threshold
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_senior_citizen_credit.py:25]: any implies tax unit
```suggestion
        elderly = age >= p.age_threshold
```

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_senior_citizen_credit.py [hunk 3]
```diff
@@ -20,11 +20,12 @@ def formula(tax_unit, period, parameters):
...
+        elderly_eligible = tax_unit.sum(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_senior_citizen_credit.py:26]:
```suggestion
        eligible = tax_unit.sum(
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
please also add a test that would have previously failed and now passes (let's do this in general when making a fix)

**Comment by vrathi101** [comment]
> est that would have previously failed and no

The issue did not include a test that failed. Not sure where to get the integration test from.

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_senior_citizen_credit.py [hunk 4]
```diff
@@ -20,11 +20,10 @@ def formula(tax_unit, period, parameters):
         has_not_taken_lump_sum_distribution = person(
             "oh_has_not_taken_oh_lump_sum_credits", period
         )
-        age = person("age", period)
 
-        any_elderly = tax_unit.any(age >= p.age_threshold)
+        age = person("age", period)
+        elderly = age >= p.age_threshold
+        eligible = tax_unit.sum(has_not_taken_lump_sum_distribution & elderly)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_senior_citizen_credit.py:26]: Please select the correct one
```suggestion
        eligible = has_not_taken_lump_sum_distribution & elderly
        count_eligible = tax_unit.sum(eligible)
        credit_amount_per_person = p.agi_limit.calc(tax_unit("oh_agi", period))
        return count_eligible * credit_amount_per_person
```
```suggestion
        eligible = has_not_taken_lump_sum_distribution & elderly
        eligible_filer = tax_unit.any(eligible)
        credit_amount = p.agi_limit.calc(tax_unit("oh_agi", period))
        return eligible_filer * credit_amount
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Yes, I understand there isn't already a test. Please add a new test that would have failed under the old (incorrect) code that will now pass.

**Review by MaxGhenis — APPROVED** [review]
Merging to stop model crashes, but in the future please adopt test-driven development

**Pull request merged**